### DISCUSSION TITLE
docker + nginx reverse proxy

### DISCUSSION THREAD

**Discussion by y4my4my4m** [mynaparrot/plugNmeet-server#348] [General]
I'd like to use it with nginx reverse proxy instead of haproxy.
Also I don't want to install it on the server but do everything via the docker

```
server {
    listen 80;
    server_name MYSITE
    return 301 https://$host$request_uri; # Redirect all HTTP requests to HTTPS
}

server {
    listen 443 ssl;
    server_name MYSITE

    ssl_certificate /etc/letsencrypt/live/MYSITE/fullchain.pem; # Path to your Let's Encrypt fullchain.pem
    ssl_certificate_key /etc/letsencrypt/live/MYSITE/privkey.pem; # Path to your Let's Encrypt privkey.pem

    location / {
        proxy_pass http://127.0.0.1:8080; # Pointing to the first plugNmeet-server IP as an example
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

    }

}
```

However, i get this error:
```
Error
Error: could not establish signal connection: room connection has timed out (signal)
```

Am I missing configurations?
I've confirmed that the required ports are opened...

I've also tried changing:
`/pkg/controller/auth_token.go`

to
```
livekitHost := strings.Replace(config.AppCnf.LivekitInfo.Host, "host.docker.internal", "MYSITE", 1) // without this you won't be able to connect
```
to no avail...

Any help?

**Comment by jibon57** [comment]
The error you mentioned related with livekit. Your browser can't established connection with livekit properly. I ain't sure if nginx support tcp pass-through or not. If not then nginx can't be use in this case.

**Comment by y4my4my4m** [comment]
@jibon57 can't i direct to IP connect? or does it need SSL ?

**Reply by jibon57** [reply]
As this isn't an issue, I moved to discussions. 
Yes, for production environment SSL is must.

**Comment by WoozyFace** [comment]
Hi, sorry to revive this old topic, but I’m running into a similar situation.
I’m installing plugNmeet on Debian, but I already have multiple servers behind a central reverse proxy. On this host, ports 80 and 443 (and the media ports) are already in use by other services, and Let’s Encrypt is handled on a different proxy server.
Every time I run the install script, it fails at the Let’s Encrypt step because it expects to bind directly to port 80 on the plugNmeet machine. In my setup, I’d actually prefer to terminate TLS and manage certificates on my existing proxy, and then just forward the traffic to the local plugNmeet instance.
Is there a way to:
disable the built-in Let’s Encrypt / HAProxy part of the installer, and
run plugNmeet behind my own reverse proxy using my existing certificates?
Or alternatively, is there a documented manual install method (without the automatic HAProxy/Let’s Encrypt integration) for this kind of setup?
Thanks in advance!
